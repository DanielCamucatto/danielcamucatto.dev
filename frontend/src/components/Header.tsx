
import { getTranslations } from '../utils/translations';

interface HeaderTranslations {
  name: string;
  role: string;
  description: string;
}

export default function Header() {
  const translations = getTranslations('pt').header as HeaderTranslations;

  return (
    <header className="lg:sticky lg:top-0 lg:max-h-screen lg:w-1/2 lg:flex lg:flex-col lg:justify-between lg:py-24">
      <div>
        <div className="mb-4">
          <h1 className="text-4xl font-bold tracking-tight text-slate-800 dark:text-slate-200 sm:text-5xl">
            {translations.name}
          </h1>
          <h2 className="mt-3 text-lg font-medium tracking-tight text-slate-700 dark:text-slate-200 sm:text-xl">
            {translations.role}
          </h2>
          <p className="mt-4 max-w-xs leading-normal text-slate-600 dark:text-slate-400">
            {translations.description}
          </p>
        </div>
      </div>
    </header>
  );
}