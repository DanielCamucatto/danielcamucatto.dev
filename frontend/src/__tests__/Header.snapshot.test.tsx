import { render } from '@testing-library/react';
import Header from '../components/Header';
describe('Header snapshot', () => {
  it('deve bater o snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
