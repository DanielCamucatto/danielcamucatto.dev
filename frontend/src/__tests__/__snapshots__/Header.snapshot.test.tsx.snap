// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header snapshot deve bater o snapshot 1`] = `
<DocumentFragment>
  <header
    class="lg:sticky lg:top-0 lg:max-h-screen lg:w-1/2 lg:flex lg:flex-col lg:justify-between lg:py-24"
  >
    <div>
      <div
        class="mb-4"
      >
        <h1
          class="text-4xl font-bold tracking-tight text-slate-800 dark:text-slate-200 sm:text-5xl"
        >
          Daniel Santana Camu√ßatto
        </h1>
        <h2
          class="mt-3 text-lg font-medium tracking-tight text-slate-700 dark:text-slate-200 sm:text-xl"
        >
          Desenvolvedor Full Stack
        </h2>
        <p
          class="mt-4 max-w-xs leading-normal text-slate-600 dark:text-slate-400"
        >
          Sou apaixonado por tecnologia, e ideias disruptivas.
        </p>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
